test/test_individual.py

```diff
@@ -1,139 +1,165 @@
 import math
+import numpy as np
 import pickle
+from collections import namedtuple
+
 import pytest
 
 import cgp
-from cgp.individual import IndividualSingleGenome
+from cgp import IndividualSingleGenome, IndividualMultiGenome
 from cgp.genome import ID_INPUT_NODE, ID_OUTPUT_NODE, ID_NON_CODING_GENE
 
 
-def test_pickle_individual():
+TestParams = namedtuple("TestParams", ["genome_params", "primitives", "dna", "target_function"])
+params_list = [
+    TestParams(
+        genome_params={
+            "n_inputs": 1,
+            "n_outputs": 1,
+            "n_columns": 2,
+            "n_rows": 1,
+            "levels_back": 2,
+        },
+        primitives=(cgp.Add, cgp.Parameter),
+        dna=[
+            ID_INPUT_NODE,
+            ID_NON_CODING_GENE,
+            ID_NON_CODING_GENE,
+            1,
+            0,
+            0,
+            0,
+            0,
+            1,
+            ID_OUTPUT_NODE,
+            2,
+            ID_NON_CODING_GENE,
+        ],
+        target_function=lambda x, c: x + c,
+    )
+]
+FuncArgs = namedtuple("FuncArgs", ["x", "c"])
+
+func_args_list = [FuncArgs(x=[3.0, 5.0], c=1.0), FuncArgs(x=[3.0], c=2.0)]
+
+
+def _create_genome(genome_params, primitives, dna):
+    genome = cgp.Genome(**genome_params, primitives=primitives)
+    genome.dna = dna
+    return genome
+
+
+def _create_individual(fitness, genome, individual_type="SingleGenome"):
+    if individual_type == "SingleGenome":
+        return IndividualSingleGenome(fitness, genome)
+    elif individual_type == "MultiGenome":
+        return IndividualMultiGenome(fitness, [genome])
+    else:
+        raise NotImplementedError
+
+
+def _unpack_evaluation(value, individual_type="SingleGenome"):
+    if individual_type == "SingleGenome":
+        return value
+    elif individual_type == "MultiGenome":
+        return value[0]
+    else:
+        raise NotImplementedError
+
+
+def _unpack_genome(individual, individual_type="SingleGenome"):
+    if individual_type == "SingleGenome":
+        return individual.genome
+    elif individual_type == "MultiGenome":
+        return individual.genome[0]
+    else:
+        raise NotImplementedError
+
+
+@pytest.mark.parametrize("individual_type", ["SingleGenome", "MultiGenome"])
+def test_pickle_individual(individual_type):
 
     primitives = (cgp.Add,)
     genome = cgp.Genome(1, 1, 1, 1, 1, primitives)
-    individual = IndividualSingleGenome(None, genome)
+    individual = _create_individual(None, genome, individual_type)
```

Comment:
maybe you could provide a default value for `fitness` s/t you don't need to provide `None`?

Reply:
why? I think it would be misleading to put a value here, if it is no needed.

Reply:
sorry, my comment wasn't clear enough: i mean in the `_create_individual` function you could provide a default value such that you don't need to provide anything here

Reply:
Yes, good idea, added that in 7772699715c2e633bdace8fda363eefe9f80dca9